src/main/java/com/couchbase/lite/router/URLConnection.java

```diff
@@ -47,12 +45,8 @@
 
     public URLConnection(URL url) {
         super(url);
-        responseInputStream = new PipedInputStream();
-        try {
-            responseOutputStream = new PipedOutputStream((PipedInputStream)responseInputStream);
-        } catch (IOException e) {
-            Log.e(Log.TAG_ROUTER, "Exception creating piped output stream", e);
-        }
+        responseOutputStream = new BufferOutputStream();
+        responseInputStream = new BufferInputStream((BufferOutputStream)responseOutputStream);
```

Comment:
I understand `PipedInputStream` and `PipedOutputStream` causes deadlock if they are used from single thread.

Can we just use [BufferedInputStream](http://docs.oracle.com/javase/6/docs/api/java/io/BufferedInputStream.html) and [BufferedOutputStream](http://docs.oracle.com/javase/6/docs/api/java/io/BufferedOutputStream.html) instead of writing `BufferInputStream` and `BufferOutputStream `?